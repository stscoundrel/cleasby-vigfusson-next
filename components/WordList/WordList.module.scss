.list {
  columns: 2;
}

@media only screen and (min-width: 992px) {
  .list {
    columns: 3;
  }
}

@media only screen and (min-width: 1280px) {
  .list {
    columns: 4;
  }
}