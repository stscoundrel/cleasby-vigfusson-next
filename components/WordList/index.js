import WordLink from 'components/WordLink'
import styles from './WordList.module.scss'

export default function WordList({ words }) {
  return (
    <ul className={styles.list}>
      { words.map((word) => (
        <li key={word.slug}>
          <WordLink data={word} />
        </li>
      )) }
    </ul>
  )
}
