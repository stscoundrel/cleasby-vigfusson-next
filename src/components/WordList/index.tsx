import { DictionaryEntry } from 'cleasby-vigfusson-dictionary'
import WordLink from 'components/WordLink'
import { DictionaryEntryDTO } from 'lib/services/dictionary'
import { hasProperty } from 'spyrjari'
import styles from './WordList.module.scss'

interface WordListProps {
  words: DictionaryEntryDTO[] | DictionaryEntry[],
  showDefinition?: boolean
}

export default function WordList({ words, showDefinition = false }: WordListProps) {
  return (
    <ul className={styles.list}>
      { words.map((word) => (
        <li key={word.slug}>
          <WordLink data={word} />
          {showDefinition && hasProperty(word, 'definitions') && <p dangerouslySetInnerHTML={{
            __html: word.definitions[0],
          } } />}
        </li>
      )) }
    </ul>
  )
}
