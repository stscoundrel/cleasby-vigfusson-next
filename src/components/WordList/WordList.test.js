import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import WordList from './index'

const words = [
  {
    word: 'af-burðr',
    definitions: [
      'm. (also spelt abb-), <i>odds, balance, bias, success</i> (cp. bera af, <i>to prevail</i>); kvað honum eigi annat vænna til afburðar, <i>in order to get the better of it,</i> Sd. 166; sá hann at engi varð afburðrinn, <i>they fought ‘aequo Marte,’</i> Sturl. ii. 74; hann ætlaði sér afburð, <i>he meant to keep the odds in his own hand,</i> Ísl. ii. 450; skal nú faraí haustvíking, ok vilda ek, at hon yrði eigi með minnum afburðum, <i>less glorious,</i> Orkn. 464.',
      '<strong>II.</strong> gen. sing. and pl. <strong>afburðar-, a-,</strong> freq. used as a prefix in some COMPDS with the notion of <i>gloriously, with distinction.</i> <strong>afburðar-digr,</strong> adj. <i>very thick</i>, biðr. 24. <strong>afburða-fræknligr,</strong> adj. <i>very gallant,</i> Ísl. ii. 369. <strong>afburðar-járn,</strong> n. <i>excellent iron,</i> Fms. x. 173. <strong>afburðar-maðr,</strong> m. <i>a man of mark,</i> Rb. 316, Orkn. 474, Grett. 133, Finnb. 318. <strong>afburðar-mikill,</strong> adj. <i>conspicuous,</i> Fms. v. 181. <strong>afburðar-skip,</strong> n. <i>a fine ship,</i> Fas. iii. 106. <strong>afburðar-vel,</strong> adv. <i>very well</i>, Hkr. ii. 265, Fms. ix. 515. <strong>afburðar-vænn,</strong> adj. <i>very fine,</i> Fas. i. 182.',
    ],
    slug: 'af-burdr',
  },
  {
    word: 'af-búð',
    definitions: [
      'f. <i>an ‘off-booth,’ side-booth, apartment,</i> Korm. 116.',
    ],
    slug: 'af-bud',
  },
  {
    word: 'af-dalr',
    definitions: [
      'm. <i>an ‘off-dale,’ remote valley;</i> freq. in tales and rhymes of <i>hidden valleys,</i> esp. in pl., e. g. Hvað hét hundr karls er í afdölum bjó, in a nursery rhyme, K. Þ. K. 38, Fms. v. 183.',
    ],
    slug: 'af-dalr',
  },
  {
    word: 'af-deilingr',
    definitions: [
      'm. <i>part, portion, share,</i> Bs. i. 881.',
    ],
    slug: 'af-deilingr',
  },
  {
    word: 'af-dráttr',
    definitions: [
      'm. [draga af, <i>to detract</i>], <i>diminution, deduction,</i> Ann. 1358 (of <i>duties, fines</i>), Dipl. i. 7, Jm. 135 <i>= costs.</i>',
      '<strong>β.</strong> in arithmetic, <i>subtraction,</i> Alg. 358, now frádragning.',
    ],
    slug: 'af-drattr',
  },
]

describe('WordList component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<WordList words={words} />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<WordList words={words} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Renders correct amount of words', () => {
    const tree = renderer.create(<WordList words={words} />)
    const { root } = tree

    expect(root.findAllByType('ul').length).toEqual(1)
    expect(root.findAllByType('li').length).toEqual(5)
  })
})
