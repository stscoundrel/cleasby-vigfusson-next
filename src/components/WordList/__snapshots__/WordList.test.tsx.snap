// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordList component Matches snapshot (with definitions) 1`] = `
<ul
  className="list"
>
  <li>
    <a
      className="link"
      href="/word/af-burdr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      af-burðr
    </a>
    <p
      dangerouslySetInnerHTML={
        {
          "__html": "m. (also spelt abb-), <i>odds, balance, bias, success</i> (cp. bera af, <i>to prevail</i>); kvað honum eigi annat vænna til afburðar, <i>in order to get the better of it,</i> Sd. 166; sá hann at engi varð afburðrinn, <i>they fought ‘aequo Marte,’</i> Sturl. ii. 74; hann ætlaði sér afburð, <i>he meant to keep the odds in his own hand,</i> Ísl. ii. 450; skal nú faraí haustvíking, ok vilda ek, at hon yrði eigi með minnum afburðum, <i>less glorious,</i> Orkn. 464.",
        }
      }
    />
  </li>
  <li>
    <a
      className="link"
      href="/word/af-bud"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      af-búð
    </a>
    <p
      dangerouslySetInnerHTML={
        {
          "__html": "f. <i>an ‘off-booth,’ side-booth, apartment,</i> Korm. 116.",
        }
      }
    />
  </li>
  <li>
    <a
      className="link"
      href="/word/af-dalr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      af-dalr
    </a>
    <p
      dangerouslySetInnerHTML={
        {
          "__html": "m. <i>an ‘off-dale,’ remote valley;</i> freq. in tales and rhymes of <i>hidden valleys,</i> esp. in pl., e. g. Hvað hét hundr karls er í afdölum bjó, in a nursery rhyme, K. Þ. K. 38, Fms. v. 183.",
        }
      }
    />
  </li>
  <li>
    <a
      className="link"
      href="/word/af-deilingr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      af-deilingr
    </a>
    <p
      dangerouslySetInnerHTML={
        {
          "__html": "m. <i>part, portion, share,</i> Bs. i. 881.",
        }
      }
    />
  </li>
  <li>
    <a
      className="link"
      href="/word/af-drattr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      af-dráttr
    </a>
    <p
      dangerouslySetInnerHTML={
        {
          "__html": "m. [draga af, <i>to detract</i>], <i>diminution, deduction,</i> Ann. 1358 (of <i>duties, fines</i>), Dipl. i. 7, Jm. 135 <i>= costs.</i>",
        }
      }
    />
  </li>
</ul>
`;

exports[`WordList component Matches snapshot 1`] = `
<ul
  className="list"
>
  <li>
    <a
      className="link"
      href="/word/af-burdr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      af-burðr
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/af-bud"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      af-búð
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/af-dalr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      af-dalr
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/af-deilingr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      af-deilingr
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/af-drattr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      af-dráttr
    </a>
  </li>
</ul>
`;
